pyspecdata/core.py

```diff
@@ -5458,6 +5458,14 @@ def extend(self,axis,extent, fill_with=0, tolerance=1e-5):
         assert du > 0, thismsg# ascending
         start_index = 0
         stop_index = len(u)
+        stop_index = extent 
+        logger.debug("*** *** *** *** ***")
+        logger.debug(extent)
+        logger.debug(u[-1])
+        logger.debug(du)
+        logger.debug(extent*du-u[-1])
+        logger.debug(int((extent - u[-1]) // du))
+        logger.debug("*** *** *** *** ***")
```

Comment:
in general, try to use logger statements -- I know about the file writing issue, but I have a patch for that that I just want @guinneak198 to review